File: Java/Medium/CompileError/all-ancestors-of-a-node-in-a-directed-acyclic-graph_3.java
class Solution {
    public List<List<Integer>> getAncestors(int n, int[][] edges) {
        
        List<List<Integer>> ancestors = new ArrayList<>();
        for (int i = 0; i < n; i++) {
            ancestors.add(new ArrayList<>());
        }
        Map<Integer, List<Integer>> graph = new HashMap<>();
        for (int i = 0; i < n; i++) {
            graph.put(i, new ArrayList<>());
        }
        for (int[] edge : edges) {
            graph.get(edge[1]).add(edge[0]);
        }
        for (int i = 0; i < n; i++) {
            dfs(graph, i, ancestors);
        }
        return ancestors;
    }

    private void dfs(Map<Integer, List<Integer>> graph, int node, List<List<Integer>> ancestors) {
        for (int neighbor : graph.get(node)) {
            ancestors.get(node).add(neighbor);
            dfs(graph, neighbor, ancestors);
            ancestors.get(node).addAll(ancestors.get(neighbor));
        }
    }
}
    }
}

File: C++/Medium/CompileError/all-ancestors-of-a-node-in-a-directed-acyclic-graph_3.cpp
class Solution {
public:
    vector<vector<int>> getAncestors(int n, vector<vector<int>>& edges) {
        
        vector<vector<int>> graph(n);
        for (auto& edge: edges) {
            graph[edge[1]].push) {
            adj[edge[0]].push_back(edge[1]);
            indegree[edge[1]]++;
        }
        vector<vector<int>> ans(n);
        queue<int> q;
        for (int i = 0; i < n; i++) {
            if (indegree[i] == 0) {
                q.push(i);
            }
        }
        while (!q.empty()) {
            int node = q.front();
            q.pop();
            for (int child : adj[node]) {
                ans[child].push_back(node);
                for (int ancestor : ans[node]) {
                    ans[child].push_back(ancestor);
                }
                sort(ans[child].begin(), ans[child].end());
                indegree[child]--;
                if (indegree[child] == 0) {
                    q.push(child);
                }
            }
        }
        return ans;
    }
    vector<vector<int>> allAncestors(int n, vector<vector<int>>& edges) {
        vector<vector<int>> ans;
        for (int i = 0; i < n; i++) {
            ans.push_back({});
        }
        vector<vector<int>> adj(n);
        vector<int> indegree(n, 0);
        for (auto &edge : edges) {
            adj[edge[0]].push_back(edge[1]);
            indegree[edge[1]]++;
        }
        queue<int> q;
        for (int i = 0; i < n; i++) {
            if (indegree[i] == 0) {
                q.push(i);
            }
        }
        while (!q.empty()) {
            int node = q.front();
            q.pop();
            for (int child : adj[node]) {
                ans[child].push_back(node);
                for (int ancestor : ans[node]) {
                    ans[child].push_back(ancestor);
                }
                sort(ans[child].begin(), ans[child].end());
                indegree[child]--;
                if (indegree[child] == 0) {
                    q.push(child);
                }
            }
        }
        return ans;
    }
};